### PR TITLE
tidb-in-kubernetes: add TiDB lightning deployment guide in kubernetes

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? 

add TiDB lightning deployment guide in kubernetes

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs/pull/1481

### Which version does your change affect? 

dev, v3.0

I will update `v3.0` after the review on `dev` is done.

### CODE REVIEW

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+category: how-to
+---
+
+# 恢复 Kubernetes 上的 TiDB 集群数据
+
+本文介绍了如何使用 [TiDB Lightning](https://github.com/pingcap/tidb-lightning) 快速恢复 Kubernetes 上的 TiDB 集群数据。
+
+TiDB Lightning 包含两个组件：tidb-lightning 和 tikv-importer。在 Kubernetes 上，tikv-importer 位于 TiDB 集群的 Helm 图表内，被部署为一个副本数为 1 (`replicas=1`) 的 `StatefulSet`；tidb-lightning 位于单独的 Helm 图表内，被部署为一个 `Job`。
+
+为了使用 TiDB Lightning 恢复数据，tikv-importer 和 tidb-lightning 都必须分别部署，部署方式如下。
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
为了使用 TiDB Lightning 恢复数据，tikv-importer 和 tidb-lightning 都必须分别部署。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+title: 恢复 Kubernetes 上的 TiDB 集群数据
+summary: 使用 TiDB Lightning 快速恢复 Kubernetes 上的 TiDB 集群数据。
+category: how-to
+---
+
+# 恢复 Kubernetes 上的 TiDB 集群数据
+
+本文介绍了如何使用 [TiDB Lightning](https://github.com/pingcap/tidb-lightning) 快速恢复 Kubernetes 上的 TiDB 集群数据。
+
+TiDB Lightning 包含两个组件：tidb-lightning 和 tikv-importer。在 Kubernetes 上，tikv-importer 位于 TiDB 集群的 Helm 图表内，被部署为一个副本数为 1 (`replicas=1`) 的 `StatefulSet`；tidb-lightning 位于单独的 Helm 图表内，被部署为一个 `Job`。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
TiDB Lightning 包含两个组件：tidb-lightning 和 tikv-importer。在 Kubernetes 上，tikv-importer 位于 TiDB 集群的 Helm chart 内，被部署为一个副本数为 1 (`replicas=1`) 的 `StatefulSet`；tidb-lightning 位于单独的 Helm chart 内，被部署为一个 `Job`。
```

"helm chart" is a terminology

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+为了使用 TiDB Lightning 恢复数据，tikv-importer 和 tidb-lightning 都必须分别部署，部署方式如下。
+
+## 部署 tikv-importer
+
+tikv-importer 可以在一个现有的 TiDB 集群上启用，或者在新建 TiDB 集群时启用。
+
+* 新建一个 TiDB 集群时启用 tikv-importer：
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
* 在新建一个 TiDB 集群时启用 tikv-importer：
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    使用如下命令获得 TiDB Lightning 的默认配置。
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    helm inspect values pingcap/tidb-lightning --version=<chart-version> > tidb-lightning-values.yaml
+    ```
+
+    TiDB Lightning Helm 图表支持本地和远程的数据源。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
    tidb-lightning Helm chart 支持恢复本地或远程的备份数据。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+    * 本地
+
+        本地模式要求 Mydumper 备份数据位于其中一个 Kubernetes 节点上。为了启用该模式，你需要将 `dataSource.local.nodeName` 设置为该节点名称，将 `dataSource.local.hostPath` 设置为 Mydumper 备份数据目录路径，该路径中包含一个名为 `metadata` 的文件。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
        本地模式要求 Mydumper 备份数据位于其中一个 Kubernetes 节点上。为了启用该模式，你需要将 `dataSource.local.nodeName` 设置为该节点名称，将 `dataSource.local.hostPath` 设置为 Mydumper 备份数据目录路径，该路径中需要包含名为 `metadata` 的文件。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+    * 远程
+
+        与本地模式不同，远程模式需要使用 [rclone](https://rclone.org) 将 Mydumper 备份 tarball 文件从网络存储中下载到 PV 中。该操作对于 rclone 支持的任何云存储应该都可行，但目前只有如下几种经过测试：[Google Cloud Storage (GCS)](https://cloud.google.com/storage/)、[AWS S3](https://aws.amazon.com/s3/) 和 [Ceph Object Storage](https://ceph.com/ceph-storage/object-storage/)。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
        与本地模式不同，远程模式需要使用 [rclone](https://rclone.org) 将 Mydumper 备份 tarball 文件从网络存储中下载到 PV 中。远程模式能在 rclone 支持的任何云存储下工作，目前已经有以下存储进行了相关测试：[Google Cloud Storage (GCS)](https://cloud.google.com/storage/)、[AWS S3](https://aws.amazon.com/s3/) 和 [Ceph Object Storage](https://ceph.com/ceph-storage/object-storage/)。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+        1. 确保 `dataSource.local.nodeName` 和 `dataSource.local.hostPath` 被注释掉。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
        1. 确保 `values.yaml` 中的 `dataSource.local.nodeName` 和 `dataSource.local.hostPath` 被注释掉。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+              endpoint = <ceph-object-store-endpoint>
+              region = :default-placement
+              [gcs]
+              type = google cloud storage
+              # The service account must include Storage Object Viewer role
+              # The content can be retrieved by `cat <service-account-file.json> | jq -c .`
+              service_account_credentials = <service-account-json-file-content>
+            ```
+
+            使用你的实际配置替换上述配置中的占位文字，并将该文件存储为 `secret.yaml`。然后通过 `kubectl apply -f secret.yaml -n <namespace>` 命令创建 `Secret`。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
            使用你的实际配置替换上述配置中的占位符，并将该文件存储为 `secret.yaml`。然后通过 `kubectl apply -f secret.yaml -n <namespace>` 命令创建该 `Secret`。
```

“占位符” is a terminology

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+2. 部署 TiDB Lightning
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    helm install pingcap/tidb-lightning --name=<tidb-lightning-release-name> --namespace=<namespace> --set failFast=true -f tidb-lightning-values.yaml --version=<chart-version>
+    ```
+
+当 TiDB Lightning 未能成功恢复数据时，它不能简单地直接重启，必须要进行**手动干预**，否则很容易出现错误。因此 tidb-lightning 的 `Job` 重启策略被设置为 `Never`。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
当 TiDB Lightning 未能成功恢复数据时，不能简单地直接重启进程，必须进行**手动干预**，否则将很容易出现错误。因此，tidb-lightning 的 `Job` 重启策略被设置为 `Never`。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md (hunk 5 above)
Comment (dcalvin) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
        本地模式要求 Mydumper 备份数据位于其中一个 Kubernetes 节点上。要启用该模式，你需要将 `dataSource.local.nodeName` 设置为该节点名称，将 `dataSource.local.hostPath` 设置为 Mydumper 备份数据目录路径（该路径下包含一个名为 `metadata` 的文件）。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    helm inspect values pingcap/tidb-lightning --version=<chart-version> > tidb-lightning-values.yaml
+    ```
+
+    TiDB Lightning Helm 图表支持本地和远程的数据源。
+
+    * 本地
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
    * 本地模式
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+
+        本地模式要求 Mydumper 备份数据位于其中一个 Kubernetes 节点上。为了启用该模式，你需要将 `dataSource.local.nodeName` 设置为该节点名称，将 `dataSource.local.hostPath` 设置为 Mydumper 备份数据目录路径，该路径中包含一个名为 `metadata` 的文件。
+
+    * 远程
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
    * 远程模式
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+
+    ```shell
+    helm install pingcap/tidb-lightning --name=<tidb-lightning-release-name> --namespace=<namespace> --set failFast=true -f tidb-lightning-values.yaml --version=<chart-version>
+    ```
+
+当 TiDB Lightning 未能成功恢复数据时，它不能简单地直接重启，必须要进行**手动干预**，否则很容易出现错误。因此 tidb-lightning 的 `Job` 重启策略被设置为 `Never`。
+
+> **注意：**
+>
+> 目前，即使数据被成功恢复，TiDB Lightning 也会[报出非零错误码并退出](https://github.com/pingcap/tidb-lightning/pull/230)，这会触发作业的失败。因此，数据恢复成功与否需要通过查看 tidb-lightning pod 的日志进行确认。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
> 目前，即使数据被成功恢复，TiDB Lightning 也会[报出非零错误码并退出](https://github.com/pingcap/tidb-lightning/pull/230)，这会导致 Job 失败。因此，数据恢复成功与否需要通过查看 tidb-lightning pod 的日志进行确认。
```

Kubernetes *Job* is a terminology

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+
+如果 TiDB Lightning 未能成功恢复数据，需要采用以下步骤进行手动干预：
+
+1. 运行 `kubectl delete job -n <namespace> <tidb-lightning-release-name>-tidb-lightning`，删除 lightning 作业。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
1. 运行 `kubectl delete job -n <namespace> <tidb-lightning-release-name>-tidb-lightning`，删除 lightning Job。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+
+2. 运行 `helm template pingcap/tidb-lightning --name <tidb-lightning-release-name> --set failFast=false -f tidb-lightning-values.yaml | kubectl apply -n <namespace> -f -`，重新创建将 `failFast` 禁用的 lightning 作业。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
2. 运行 `helm template pingcap/tidb-lightning --name <tidb-lightning-release-name> --set failFast=false -f tidb-lightning-values.yaml | kubectl apply -n <namespace> -f -`，重新创建将 `failFast` 禁用的 lightning Job。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+
+3. 当 lightning pod 重新运行时，在 lightning 容器中执行 `kubectl exec -it -n <namesapce> <tidb-lightning-pod-name> sh` 命令。
+
+4. 运行 `cat /proc/1/cmdline`，获得启动脚本。
+
+5. 阅读[故障排除指南](/how-to/troubleshoot/tidb-lightning.md#tidb-lightning-troubleshooting)，对 lightning 进行诊断。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
5. 参考[故障排除指南](/how-to/troubleshoot/tidb-lightning.md#tidb-lightning-troubleshooting)，对 lightning 进行诊断。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+
+## 销毁 TiDB Lightning
+
+目前，TiDB Lightning 只能在线下恢复数据。当恢复过程结束、TiDB 集群需要向外部应用提供服务时，需要销毁 TiDB Lightning 以节省开支。
```
Comment (aylei) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
目前，TiDB Lightning 只能在线下恢复数据。当恢复过程结束、TiDB 集群需要向外部应用提供服务时，可以销毁 TiDB Lightning 以节省开支。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+
+        本地模式要求 Mydumper 备份数据位于其中一个 Kubernetes 节点上。为了启用该模式，你需要将 `dataSource.local.nodeName` 设置为该节点名称，将 `dataSource.local.hostPath` 设置为 Mydumper 备份数据目录路径，该路径中包含一个名为 `metadata` 的文件。
+
+    * 远程
+
+        与本地模式不同，远程模式需要使用 [rclone](https://rclone.org) 将 Mydumper 备份 tarball 文件从网络存储中下载到 PV 中。该操作对于 rclone 支持的任何云存储应该都可行，但目前只有如下几种经过测试：[Google Cloud Storage (GCS)](https://cloud.google.com/storage/)、[AWS S3](https://aws.amazon.com/s3/) 和 [Ceph Object Storage](https://ceph.com/ceph-storage/object-storage/)。
+
+        1. 确保 `dataSource.local.nodeName` 和 `dataSource.local.hostPath` 被注释掉。
+
+        2. 新建一个包含 rclone 配置的 `Secret`。rclone 配置实例如下所示。只要求配置一种云存储。有关其他的云存储，请参考 [rclone 官方文档](https://rclone.org/)。
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
        2. 新建一个包含 rclone 配置的 `Secret`。配置示例如下。示例中只要求配置一种云存储，有关其他的云存储，请参考 [rclone 官方文档](https://rclone.org/)。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+              type = s3
+              provider = Ceph
+              env_auth = false
+              access_key_id = <my-access-key>
+              secret_access_key = <my-secret-key>
+              endpoint = <ceph-object-store-endpoint>
+              region = :default-placement
+              [gcs]
+              type = google cloud storage
+              # The service account must include Storage Object Viewer role
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/maintain/lightning.md]: should be translated?

File: dev/tidb-in-kubernetes/maintain/lightning.md (hunk 14 above)
Comment (dcalvin) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
2. 运行 `helm template pingcap/tidb-lightning --name <tidb-lightning-release-name> --set failFast=false -f tidb-lightning-values.yaml | kubectl apply -n <namespace> -f -`，重新创建禁用 `failFast` 的 lightning 作业。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md (hunk 15 above)
Comment (dcalvin) [inline dev/tidb-in-kubernetes/maintain/lightning.md]:
```suggestion
5. 按照[故障排除指南](/how-to/troubleshoot/tidb-lightning.md#tidb-lightning-troubleshooting)，对 lightning 进行诊断。
```

File: dev/tidb-in-kubernetes/maintain/lightning.md (hunk 17 above)
Comment (yikeke) [reply dev/tidb-in-kubernetes/maintain/lightning.md]: 这里是“下面的示例只要求配置一种云存储”（个例），还是“rclone 配置中只要求配置一种云存储”（一般情况）呢？@tennix @aylei

Comment (tennix) [reply dev/tidb-in-kubernetes/maintain/lightning.md]: 备份需要往哪里上传就配置哪种云存储，一般只会上传到一种云存储里面。

Comment (yikeke) [reply dev/tidb-in-kubernetes/maintain/lightning.md]: Updated. PTAL again. @dcalvin

**Pull request merged**